### PR TITLE
Stop spell check on push

### PR DESCRIPTION
**Pull request by VictoriousRaptor**
Stop checking on push to avoid checking merge commits on dev branch. Commits of PRs can be checked on synchronized.

### CODE REVIEW

File: .github/workflows/spelling.yml [hunk 1]
```diff
@@ -65,7 +65,7 @@ jobs:
     outputs:
       followup: ${{ steps.spelling.outputs.followup }}
     runs-on: ubuntu-latest
-    if: (contains(github.event_name, 'pull_request') && github.head_ref != 'l10n_dev') || github.event_name == 'push'
+    if: (contains(github.event_name, 'pull_request') && github.head_ref != 'l10n_dev')
```
Comment (jjw24) [inline .github/workflows/spelling.yml:68]: How does this pull request event work, would it only get triggered when a pr is created? If the case then it probably wouldn't be triggered again by subsequent pushes to the pr.

I wonder maybe a better way is to just  github.head_ref != dev && github.head_ref != master, so it doesn't check pushes to dev and master branches.

File: .github/workflows/spelling.yml [hunk 2]
```diff
@@ -103,21 +103,21 @@ jobs:
...
+#     permissions:
+#       contents: write
+#     if: (success() || failure()) && needs.spelling.outputs.followup && github.event_name == 'push'
+#     steps:
+#     - name: comment
+#       uses: check-spelling/check-spelling@main
+#       with:
+#         checkout: true
+#         spell_check_this: check-spelling/spell-check-this@main
+#         task: ${{ needs.spelling.outputs.followup }}
```
Comment (jjw24) [inline .github/workflows/spelling.yml:120]: Why we commenting these out?

File: .github/workflows/spelling.yml (hunk 1 above)
Comment (VictoriousRaptor) [reply .github/workflows/spelling.yml:68]: It is also triggered on synchronize event (any push to the pr head triggers it).

File: .github/workflows/spelling.yml (hunk 2 above)
Comment (VictoriousRaptor) [reply .github/workflows/spelling.yml:120]: This is bot comment for push checks. Check on push is disabled so this is not needed anymore.

File: .github/workflows/spelling.yml (hunk 1 above)
Comment (VictoriousRaptor) [reply .github/workflows/spelling.yml:68]: simply ignoring dev or master is enough though. but if you push to a fork the check is also triggered. I want to just check in the pr.

Comment (VictoriousRaptor) [reply .github/workflows/spelling.yml:68]: oh forget to comment on push at the beginning of the file

**Review by jjw24 — APPROVED** [review]

**Pull request merged**